### PR TITLE
Add documentation to TileMode.

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
cc @abarth 

Once this lands, the next engine roll should be synchronised with landing https://github.com/flutter/assets-for-api-docs/pull/1.

File: lib/ui/painting.dart [hunk 1]
```diff
@@ -1011,7 +1050,11 @@ class Gradient extends Shader {
   /// `color` must therefore only have two entries).
   ///
   /// The behavior before `from` and after `to` is described by the `tileMode`
-  /// argument.
+  /// argument. For details, see the [TileMode] enum.
+  ///
+  /// ![](https://flutter.github.io/assets-for-api-docs/dart-ui/tile_mode_clamp_linear.png)
+  /// ![](https://flutter.github.io/assets-for-api-docs/dart-ui/tile_mode_mirror_linear.png)
+  /// ![](https://flutter.github.io/assets-for-api-docs/dart-ui/tile_mode_repeated_linear.png)
```
Comment (abarth) [inline lib/ui/painting.dart]: No alt text?

**Comment by abarth** [comment]
LGTM

**Comment by chinmaygarde** [comment]
@Hixie You changed the order of the enum values. From `Clamp`, `Repeat`, `Mirror` to `Clamp`, `Mirror`, `Repeat`. This makes them go out of sync with the enum values in the `SkShader::TileMode`. Now, all repeats are mirrored and vice versa.

**Comment by Hixie** [comment]
:-(

*cube of shame*

**Pull request merged**